src/Bearded.TD/Game/Simulation/Elements/IElementalEffect.cs

```diff
@@ -1,9 +1,25 @@
+using Bearded.TD.Game.Simulation.GameObjects;
 using Bearded.Utilities.SpaceTime;
 
 namespace Bearded.TD.Game.Simulation.Elements;
 
 interface IElementalEffect
 {
-    IElementalPhenomenon Phenomenon { get; }
     TimeSpan Duration { get; }
+
+    interface IScope
+    {
+        void ApplyTick(Instant now);
+    }
+}
+
+interface IElementalEffect<in TEffect> : IElementalEffect
+    where TEffect : IElementalEffect<TEffect>
+{
+    new interface IScope : IElementalEffect.IScope
+    {
+        void Adopt(TEffect effect, Instant now);
+    }
+
+    IScope NewScope(GameObject target);
```

Comment:
While we reduce the amount of boilerplate, we now also end up with an interface that breaks the single responsibility principle: it represents an effect + duration, while also knowing how to create its own scope. Previously the phenomenon was in charge of creating the scope and knowing about the bookkeeping, and the effect only knew what type it was and the information needed to make a unique instance. That separation of responsibilities has now disappeared. I don't know if the extra boilerplate outweighs the strict separation of responsibilities, but the separation in different types was not without reason.